pkg/domain/infra/abi/containers.go

```diff
@@ -1803,16 +1766,16 @@ func (ic *ContainerEngine) ContainerUpdate(ctx context.Context, updateOptions *e
 	if err != nil {
 		return "", err
 	}
-	ctrs, err := getContainersByContext(false, false, false, []string{updateOptions.NameOrID}, ic.Libpod)
+	containers, err := getContainers(ic.Libpod, getContainersOptions{names: []string{updateOptions.NameOrID}})
```

Comment:
why is this calling getContainers() at all? If it only need one contianer without latest option it should just use LookupContainer(),

also 
```
if len(containers) != 1 {
		return "", fmt.Errorf("container not found")
}
```
is just a misleading error

Reply:
Fair point.  Cool if I tackle that in another PR?  There is a lot to improve in this file and this code in question was pre-dates this PR.

My main goal at the moment is to improve the `podman rm` code in hope to get closer to resolving https://github.com/containers/podman/issues/15367.

Reply:
sure